### PR TITLE
Align audit_rules_sudoers_d remediation with DISA STIG content

### LINKED ISSUE #15052: DISA-alignment mismatch on audit_rules_sudoers_d — SSG pass, DISA fail (RHEL 9.2/9.4/9.6/9.8/9.9)
DISA-alignment testing shows a consistent mismatch for `audit_rules_sudoers_d`, reproduced across `oscap`, `anaconda`, and `ansible` remediation scenarios.

**Result:** SSG = `pass`, DISA = `fail`
**DISA rule ID:** `SV-258218r1210920_rule`
**Affected:** RHEL 9.2, 9.4, 9.6, 9.8, 9.9 — **not** RHEL 8.10, suggesting either the expected audit rule syntax differs between RHEL 8 and 9 STIG benchmarks, or the underlying audit rule content changed for RHEL 9 in a way that satisfies SSG but not DISA's expected form (e.g. auditctl rule field ordering, key naming, or path form for `/etc/sudoers.d`).

### PR DESCRIPTION
**Pull request by ggbecker**
#### Description:

- Change `audit_rules_sudoers_d` to watch `/etc/sudoers.d` (no trailing slash), so the remediation emits the `path=` filter form instead of `dir=`.
- Make the `audit_rules_watch` OVAL template accept both the `dir` and `path` filters, with or without a trailing slash.
- Add test scenarios to `audit_rules_sudoers_d` covering all accepted forms and the negative cases.

#### Rationale:
- Fixes #15052
- The DISA RHEL 9 STIG check for SV-258218 only accepts `-a always,exit -F arch=b64 -F path=/etc/sudoers.d -F perm=wa ...`. Our content generated the semantically equivalent `-F dir=/etc/sudoers.d/` form, so systems remediated by our content passed our own scan but failed the DISA scan.
- Aligning the remediation with the DISA content resolves this mismatch, and relaxing the OVAL check to accept both forms avoids regressing systems already remediated with the previous `dir=` form.

#### Review Hints:

- The key change is dropping the trailing slash from the `path` template var in `rule.yml`; `template.py` derives `filter_type` (`dir` vs `path`) from that slash.
- The OVAL template change is additive leniency `(dir|path)` + optional trailing slash, so it never rejects anything previously accepted; it affects all 35 rules using the template.
- Tested on rhel9 with Automatus (bash remediation): all 21 scenarios pass (8 new + 13 from the shared template).
- The DISA check accepts any (or no) `-F key=` value, so keeping `key=actions` is fine; confirmed against `U_RHEL_9_V2R9_STIG_SCAP_1-3_Benchmark`.

### CODE REVIEW

File: shared/templates/audit_rules_watch/oval.template [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
     <concat>
-      <literal_component>^\-a\s+always,exit\s+\-F\s+arch={{{ arch }}}\s+\-F\s+{{{ FILTER_TYPE }}}=</literal_component>
+      <literal_component>^\-a\s+always,exit\s+\-F\s+arch={{{ arch }}}\s+\-F\s+(dir|path)=</literal_component>
   {{% if PATH_IS_VARIABLE %}}
       <variable_component var_ref="{{{ PATH }}}"/>
   {{% else %}}
       <literal_component>{{{ PATH_ESCAPED }}}</literal_component>
   {{% endif %}}
-      <literal_component>\s+\-F\s+perm=\b([rx]*w[rx]*a[rx]*|[rx]*a[rx]*w[rx]*)\b.*$</literal_component>
+      <literal_component>/?\s+\-F\s+perm=\b([rx]*w[rx]*a[rx]*|[rx]*a[rx]*w[rx]*)\b.*$</literal_component>
```
Comment (jan-cerny) [inline shared/templates/audit_rules_watch/oval.template:10]: @ggbecker Will it accidentally start accepting paths with trailing slash where it shouldn't accept it? When the audit rule is supposed to watch a regular file (not a directory) but the audit rule contains a path with a trailing slash, will the rule fail?

Let's assume that I have the following rules in the `/etc/audit/audit.rules`, which is supposed to watch the `/etc/passwd` file, but notice the path:

```
-a always,exit -F arch=b32 -F path=/etc/passwd/ -F perm=wa -F key=identity
-a always,exit -F arch=b64 -F path=/etc/passwd/ -F perm=wa -F key=identity
```

What will be the OVAL result?

File: shared/templates/audit_rules_watch/template.py [hunk 2]
```diff
@@ -7,7 +7,13 @@ def preprocess(data, lang):
     name = ssg.utils.escape_id(os.path.basename(os.path.normpath(path)))
     data["name"] = name
     if lang == "oval":
-        data["path_escaped"] = path.replace("/", "\\/")
+        # Normalize away a trailing slash so it can be made optional in the
```
Comment (jan-cerny) [inline shared/templates/audit_rules_watch/template.py:10]: @ggbecker I consider having the new behavior accessible by a new template parameter instead of changing the existing behavior.

File: linux_os/guide/auditing/auditd_configure_rules/audit_rules_sudoers_d/rule.yml [hunk 3]
```diff
@@ -6,7 +6,7 @@ title: 'Ensure auditd Collects System Administrator Actions - /etc/sudoers.d/'
 description: |-
     At a minimum, the audit system should collect administrator actions
     for all users and root.
-    {{{ describe_audit_rules_watch("/etc/sudoers.d/", "actions") }}}
+    {{{ describe_audit_rules_watch("/etc/sudoers.d", "actions") }}}
```
Comment (jan-cerny) [inline linux_os/guide/auditing/auditd_configure_rules/audit_rules_sudoers_d/rule.yml:9]: The RHEL 10 STIG contains the trailing slash in the rule, so the description won't match the rule. https://stigaview.com/products/rhel10/v1r2/RHEL-10-500690/

**Comment by Mab879** [comment]
Can we remove the ` backported-into-stabilization` label on this one?

**Comment by ggbecker** [comment]
> Can we remove the ` backported-into-stabilization` label on this one?

removed
